src/components/Checkbox/Checkbox.js

```diff
@@ -7,62 +7,55 @@ import {
   ConsentDefinitionStyled
 } from './CheckboxStyled';
 
-class Checkbox extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
-  }
-
-  render() {
-    const {
-      children,
-      onClickFn,
-      error,
-      checked,
-      required,
-      isMyAccount,
-      className,
-      disabled,
-      isRadioButton
-    } = this.props;
-    return (
-      <CheckboxStyled
-        onClick={e => onClickFn(e, disabled)}
-        role="checkbox"
-        tabIndex="-1"
-        aria-checked="false"
+const Checkbox = ({
+  children,
+  onClickFn,
+  error,
+  checked,
+  required,
+  isMyAccount,
+  className,
+  disabled,
+  isRadioButton
+}) => {
+  return (
+    <CheckboxStyled
+      onClick={e => onClickFn(e, disabled)}
+      role="checkbox"
+      tabIndex="-1"
+      aria-checked={checked}
+      checked={checked}
+      aria-label={children}
+      className={className}
+      disabled={disabled}
+    >
+      <CheckFrameStyled
+        error={error && required && !checked}
+        tabIndex="0"
+        onKeyDown={e => {
+          return e.keyCode === 32 ? onClickFn() : null;
```

Comment:
we could omit return in arrow function:
```suggestion
          onKeyDown={e => e.keyCode === 32 ? onClickFn() : null}
```